exporter/exporterhelper/common.go

```diff
@@ -42,55 +43,74 @@ func (b *baseRequestSender) setNextSender(nextSender requestSender) {
 type obsrepSenderFactory func(obsrep *ObsReport) requestSender
 
 // Option apply changes to baseExporter.
-type Option func(*baseExporter)
+type Option interface {
+	apply(*baseExporter) error
+}
+
+var _ Option = (optionFuncNoErr)(nil)
+
+type optionFuncNoErr func(*baseExporter)
+
+func (o optionFuncNoErr) apply(be *baseExporter) error {
+	o(be)
+	return nil
+}
+
+var _ Option = (optionFunc)(nil)
+
+type optionFunc func(*baseExporter) error
+
+func (o optionFunc) apply(be *baseExporter) error {
+	return o(be)
+}
```

Comment:
Why don't just use `type Option func(*baseExporter) error`?

Reply:
I was being consistent with https://github.com/open-telemetry/opentelemetry-collector/blob/94ae7762c9458cfe199cfaa3d16a4ecc1f7330be/featuregate/registry.go#L45-L48

I have a slight preference for this instead of the `func` because it is more flexible but I am fine changing it if you prefer the `func`.

Reply:
It just seems like too much extra code. I would prefer a function or at least removing `optionFuncNoErr`.

Reply:
Oki, changed back to `func` with an `error`, we can always change back if needed